debug_module.tex

```diff
@@ -56,25 +56,46 @@ \section{Debug Module Interface (DMI)} \label{dmi}
 on this DMI, the base address of the next DM in the DMI address space is given
 in \RdmNextdm.
 
-
 The Debug Module is controlled via register accesses to its DMI address space.
 
 \section{Reset Control} \label{reset}
 
-The Debug Module controls a global reset signal, \FdmDmcontrolNdmreset
-(non-debug module reset),
-which can reset, or hold in reset, every component in the platform,
-except for the Debug Module and Debug Transport Modules.
-Exactly what is affected by this reset is implementation dependent, as long as
-it is possible to debug programs from the first instruction executed.
+There are two methods that allow a debugger to reset harts.
+\FdmDmcontrolNdmreset resets all the harts in the system, as well as all other
+parts of the system except for the Debug Module and Debug Transport Modules.
+Exactly what is affected by this reset is implementation dependent, but it
+must be possible to debug programs from the first instruction executed.
+\FdmDmcontrolHartreset resets all the currently selected harts. In this case an
+implementation may reset more harts than just the ones that are selected. The
+debugger can discover which other harts are reset (if any) by selecting them
+and checking \FdmDmstatusAnyhavereset and \FdmDmstatusAllhavereset.
+
+To perform either of these resets, the debugger first asserts the bit, and then
+clears it. The actual reset may start as soon as the bit is asserted, but is allowed
+to start a long time after the bit is deasserted. While the reset is
```

Comment:
How about "but may start an arbitrarily long time after the bit is deasserted"?  It sounds a little more formal to me.

Reply:
That language also happens to mirror language that was removed below that says that the reset itself can be arbitrarily long.  I think it would be useful to say something to that effect here so it's clear that entering and exiting reset can each take a while, not just the former.

Reply:
Added. Thanks.